src/tools/clippy/clippy_lints/src/attrs/mod.rs

```diff
@@ -467,7 +467,8 @@ impl<'tcx> LateLintPass<'tcx> for Attributes {
     fn check_item(&mut self, cx: &LateContext<'tcx>, item: &'tcx Item<'_>) {
         let attrs = cx.tcx.hir().attrs(item.hir_id());
         if is_relevant_item(cx, item) {
-            inline_always::check(cx, item.span, item.ident.name, attrs);
+            let (ident, _, _, _) = item.expect_fn();
+            inline_always::check(cx, item.span, ident.name, attrs);
         }
```

Comment:
Imo, the `expect_fn` isn't "obviously correct" without looking at the implementation details of `is_relevant_item`. I would either make `is_relevant_item` return an `Option<Ident>` (but then the name no longer makes that much sense).

I'd rather you inline `is_relevant_item` (and also `is_relevant_{impl,trait}`). All of these three functions are only called once, namely in this module. This way you can get rid of the `expect_fn` which is essentially an `unwrap` and the code becomes much clearer imo and shaves off some unnecessary LOCs.

Disclaimer tho: I'm not on the Clippy team, so I don't know if I really have a say in this. It's a bit intrusive.

Reply:
I agree the `is_relevant_*` functions are confusing. I thought about inlining like you said, but it required also moving `is_relevant_{block,expr}` and that was getting a bit invasive. So I've just made it check the item kind is a `Fn` before doing the `is_relevant_item` call.